regclient/regclient
# Add concurrency to image copy

### Fixes issue

Fixes #390.


### Describe the change

This adds concurrency to the image copy command, and removes race conditions from various APIs. This image copy itself has been significantly redesigned to check all manifests, including referrers and digest tags, before copying blobs, so that the remaining bytes for a blob copy is more accurate. This removes the need for `--force-recursive` when updating referrers and digest tags but will also cause those copies to be slower and pull more manifests when nothing has changed upstream.



### How to verify it

`regctl image copy $src $dest` will now show multiple blobs copied simultaneously and will first check all manifests to get an accurate size of the remaining blob copies.


### Changelog text

Image copy is now run with concurrency.


### Please verify and check that the pull request fulfills the following requirements
